docs/MIGRATION.md

```diff
@@ -189,21 +190,59 @@ jobs:
       - name: Create a File
         run: echo "hello from ${{ matrix.runs-on }}" > file-${{ matrix.runs-on }}.txt
       - name: Upload Artifact
--        uses: actions/upload-artifact@v3
-+        uses: actions/upload-artifact@v4
+-       uses: actions/upload-artifact@v3
++       uses: actions/upload-artifact@v4
         with:
 -         name: all-my-files
 +         name: my-artifact-${{ matrix.runs-on }}
           path: file-${{ matrix.runs-on }}.txt
-+ merge:
-+   runs-on: ubuntu-latest
-+   needs: upload
-+   steps:
-+     - name: Merge Artifacts
-+       uses: actions/upload-artifact/merge@v4
-+       with:
-+         name: all-my-files
-+         pattern: my-artifact-*
++  merge:
++    runs-on: ubuntu-latest
++    needs: upload
++    steps:
++      - name: Merge Artifacts
++        uses: actions/upload-artifact/merge@v4
++        with:
++          name: all-my-files
++          pattern: my-artifact-*
 ```
 
 Note that this will download all artifacts to a temporary directory and reupload them as a single artifact. For more information on inputs and other use cases for `actions/upload-artifact/merge@v4`, see [the action documentation](../merge/README.md).
+
+## Hidden Files
+
+By default, hidden files are ignored by this action to avoid unintentionally uploading sensitive
+information.
+
+In versions of this action before v4.4.0, these hidden files were included by default.
+
+If you need to upload hidden files, you can use the `include-hidden-files` input.
```

Comment:
Do we want to mention anything about wildcards and potentially the impact of using them?

Reply:
What's the impact of using wildcards?

Reply:
Let me rephrase to - "When using wildcards be sure that you aren't accidentally uploading hidden files that may contain secrets"